webrecorder/browsertrix
# Frontend collections beta UI

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/647



### Changes

Adds new Collections feature! The top-nav link is hidden for internal testing.

Not yet implemented (different from mockups):
- ~~Deleting collections (see API issue: https://github.com/webrecorder/browsertrix-cloud/issues/869#issuecomment-1567538340)~~ Fixed!
- ~~Navigating Replay Webpage by seed URL--no easy way of getting seed URLs and timestamps with current API~~ Not doing in favor of Replay WebPage UI

### Manual testing

1. Log in as a user with "crawler" or higher role
2. Enter in the URL `orgs/{your org}/collections`. Verify collection empty state is shown
3. Click "Create Collection". Verify available workflows are shown.
4. Click a workflow. Verify crawls in workflow are loaded into "Crawls in Collection" and can be selected and deselected.
5. Click "Enter Metadata" and enter name and description. Verify fields update as expected, with validation.
6. Click "Save Collection". Verify collection is saved and name and description renders as expected.
7. Click "Actions" -> "Edit Collection" and update collection crawls. Click "Save Crawl Selection". Verify collection saves as expected and stays on the form.
8. Click "Metadata" tab.
9. Enter a long description (a few paragraphs) and click "Save Metadata". 
10. Click "Back to [collection]". Verify description is shown and clicking "More/Less" will expand/contract the description.
11. Scroll down to "Replay". Verify Replay Webpage loads as expected.
12. Log out and log in as "viewer" role. Verify all collection editing and creation UI is hidden.

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Collections (Empty state) | **Crawler role:**<br/><img width="1167" alt="Screen Shot 2023-05-29 at 10 10 19 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/0258bd7d-e616-4d8e-afd9-724d18ee033c"> <br/>**Viewer role:**<br/><img width="470" alt="Screen Shot 2023-05-29 at 3 49 54 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/39ebab26-fa4d-4929-af83-4990751262b0"> |
| New Collection - Select Crawls (Empty Workflows) | **Crawler role:**<br/><img width="931" alt="Screen Shot 2023-05-29 at 11 00 41 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/0773e5cc-a7ae-4751-b8bb-aad9634e8a02"><br/>**Viewer role:**<br/><img width="923" alt="Screen Shot 2023-05-29 at 11 04 58 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/cdb995b0-8840-4bbc-88c9-cfec220ff9da"> |
| New Collection - Select Crawls | (See **Demo** below) |
| New Collection - Metadata | <img width="877" alt="Screen Shot 2023-05-29 at 12 25 56 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/3498fa67-f1e8-4714-ae36-6d8edf52ec43"> |
| Collections (with action menu open) |  <img width="1278" alt="Screen Shot 2023-05-29 at 3 27 07 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/5dc271dc-0987-4fca-8825-9d09aa042004"> 

#### Demo
**New Collection - Select Crawls: **

https://github.com/webrecorder/browsertrix-cloud/assets/4672952/5db91c22-15bd-41b4-9e63-c4db2ac078f8


**Collection Detail - Expand/contract description:**


https://github.com/webrecorder/browsertrix-cloud/assets/4672952/c2b1fb28-def3-4ac0-8303-b1ac5c8c13e2
